### PR TITLE
Fixes #4520

### LINKED ISSUE #4520: MP REST Client TCK - Fix excluded tests
Currently, the below tests are excluded from execution when running the MP REST Client TCK.

- [ ] [timeout](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/tck-suite.xml#L22)
- [ ] [HasConversationScopeTest.*](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/tck-suite.xml#L28)
- [ ] [HasSessionScopeTest.*](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/tck-suite.xml#L33)
- [ ] [CDIInvokeSimpleGetOperationTest.testHasDependentScopedByDefault](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/tck-suite.xml#L43)
- [ ] [HasRequestScopeTest.testHasRequestScoped](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/tck-suite.xml#L49)
- [ ] [HasAppScopeTest.testHasApplicationScoped](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/tck-suite.xml#L55)
- [ ] [HasSingletonScopeTest.testHasSingletonScoped](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/tck-suite.xml#L61)
- [ ] [InvokeWithJsonPProviderTest.*](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/tck-suite.xml#L68)
- [ ] [FeatureRegistrationTest.*](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/tck-suite.xml#L73)
- [ ] [CustomHttpMethodTest.*](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/tck-suite.xml#L80)
- [ ] [RestClientBuilderListenerTest.*](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/tck-suite.xml#L85)
- [ ] [RestClientListenerTest.*](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/tck-suite.xml#L90)
- [ ] [CDIInterceptorTest.*](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/tck-suite.xml#L97)
- [ ] [ClientHeadersFactoryTest.*](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/tck-suite.xml#L102)
- [ ] [ClientReuseTest.*](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/tck-suite.xml#L107)
- [ ] [InvokeWithBuiltProvidersTest](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/tck-suite.xml#L115) Also see https://github.com/quarkusio/quarkus/issues/2801
- [ ] [CDIInvokeWithRegisteredProvidersTest](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/tck-suite.xml#L121)
- [ ] [InvokeWithJsonBProviderTest.*](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/src/test/java/io/quarkus/tck/restclient/SkipProblematicTests.java#L23)
- [ ] [InvokeWithJsonPProviderTest.*](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/src/test/java/io/quarkus/tck/restclient/SkipProblematicTests.java#L23)
- [ ] [ClientHeaderParamTest.*](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/src/test/java/io/quarkus/tck/restclient/SkipProblematicTests.java#L24)
- [ ] [AsyncMethodTest.*](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-rest-client/src/test/java/io/quarkus/tck/restclient/SkipProblematicTests.java#L31)

### PR DESCRIPTION
**Pull request by kenfinnigan**

### CODE REVIEW

File: tcks/microprofile-metrics/rest/pom.xml [hunk 1]
```diff
@@ -31,6 +31,9 @@
                     <environmentVariables>
                         <MP_METRICS_TAGS>tier=integration</MP_METRICS_TAGS>
                     </environmentVariables>
+                    <excludes>
+                        <exclude>org.eclipse.microprofile.metrics.test.MpMetricTest</exclude>
```
Comment (jmartisk) [inline tcks/microprofile-metrics/rest/pom.xml]: This is almost the whole TCK module. Why?

Comment (kenfinnigan) [reply tcks/microprofile-metrics/rest/pom.xml]: It was only one test failing, due to the Gauge issue, but there's no way to exclude a single test in pom.xml

Comment (jmartisk) [reply tcks/microprofile-metrics/rest/pom.xml]: I'm confused how you managed to get that test to fail, I thought this was an issue only for other TCKs that were compiled against Metrics 2.1. Btw I will send a PR to upgrade to Metrics 2.2 as soon as SmallRye Metrics 2.3 gets synced into Central

**Comment by kenfinnigan** [comment]
Tbh I don’t know.

If you can figure it out that’s awesome

File: extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java [hunk 2]
```diff
@@ -232,13 +209,62 @@ public void register(RegistrationContext registrationContext) {
...
+                    processInterfaceReturnTypes(interfaceClassInfo, returnTypes);
+                }
+            }
+        }
+    }
+
+    private void processInterfaceReturnTypes(ClassInfo classInfo, Set<Type> returnTypes) {
+        for (MethodInfo method : classInfo.methods()) {
+            Type type = method.returnType();
+            if (!type.name().toString().contains("java.lang")) {
```
Comment (jmartisk) [inline extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java]: nitpick: `startsWith` would be safer  than `contains`, you never know what kind of insane person could use a package that contains `java.lang` somewhere in the middle

**Comment by jmartisk** [comment]
> Tbh I don’t know. If you can figure it out that’s awesome

It must have been some stale dependencies on your machine, I don't see any reason why this PR would break that test

**Comment by kenfinnigan** [comment]
@jmartisk just verified I get the following error:

```
[ERROR] testListsAllJson(org.eclipse.microprofile.metrics.test.MpMetricTest)  Time elapsed: 0.305 s  <<< ERROR!
com.fasterxml.jackson.core.JsonParseException: 
Illegal unquoted character ((CTRL-CHAR, code 9)): has to be escaped using backslash to be included in string value
 at [Source: (String)"{"details":"Error handling fc40c116-670a-4bef-a309-0614e667bd59-1, java.lang.NumberFormatException: ","stack":"java.lang.NumberFormatException\n	at java.math.BigDecimal.&lt;init&gt;(BigDecimal.java:497)\n	at java.math.BigDecimal.&lt;init&gt;(BigDecimal.java:383)\n	at java.math.BigDecimal.&lt;init&gt;(BigDecimal.java:809)\n	at java.math.BigDecimal.valueOf(BigDecimal.java:1277)\n	at org.glassfish.json.JsonNumberImpl.getJsonNumber(JsonNumberImpl.java:46)\n	at org.glassfish.json.JsonProviderImpl.cre"[truncated 2535 chars]; line: 1, column: 146]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1804)
```

File: extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java (hunk 2 above)
Comment (kenfinnigan) [reply extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java]: Fixed

**Comment by gsmet** [comment]
Closing, will be part of another PR.

**Comment by jmartisk** [comment]
@kenfinnigan ok, I reported https://github.com/quarkusio/quarkus/issues/5014 for that and will look what can be done to re-enable it

**Pull request closed without merging**